api/classes/base_controller.js

```diff
@@ -85,31 +85,29 @@ BaseController.prototype.getAllAsMeta = function(req, reply) {
 
 BaseController.prototype.getAllAsTar = function(req, reply) {
   var files = req.pre.records.models;
+  var tarStream = Tar.pack();
   var self = this;
 
-  function createTarStream() {
-    var archive = archiver('tar');
-
+  function startStreaming() {
     return Promise.map(files, function(file) {
       return self.Model.query({
         where: {
           id: file.get('id')
         },
         columns: ['buffer']
-      }).fetch()
-      .then(function(model) {
-        archive.append(model.get('buffer'), { name: file.get('path') });
+      }).fetch().then(function(model) {
+        tarStream.entry({ name: file.get('path') }, model.get('buffer'));
       });
     }).then(function() {
-      archive.finalize();
-
-      return archive;
+      tarStream.finalize();
     });
   }
 
+  startStreaming().catch(errors.generateErrorResponse);
```

Comment:
`startStreaming(files).catch(..)`  Let's pass `files in as a local arg instead of accessing via closure.